tests/suites/main_test.function

```diff
@@ -213,7 +213,8 @@ int check_test( int func_idx )
     int ret = DISPATCH_TEST_SUCCESS;
     TestWrapper_t fp = NULL;
 
-    if ( func_idx < (int)( sizeof(test_funcs)/sizeof( TestWrapper_t ) ) )
+    if ( func_idx >= 0 &&
+         func_idx < (int)( sizeof(test_funcs)/sizeof( TestWrapper_t ) ) )
```

Comment:
Would it be more appropriate to use an unsigned type for `func_idx` and avoid this check? When would we want to allow `func_idx` to be negative?

Reply:
Negative function id doesn't make sense. I'll rework the PR to use an unsigned value here. Also I've missed another lookup like this in this module, so I'll also fix that.